### PR TITLE
Feature/idcom 1492  gk service retrieve token after action

### PR DESCRIPTION
**Pull request by kevinhcolgan**
Update the gatekeeper service to not retrieve the gateway token after an action like issue, revoke etc. are called. Instead, the gateway token to return is created from the existing variables available in the function.

Also in this PR:
* Added optional confirmOptions to action methods
* Removed unused gatekeeper-http-client

### CODE REVIEW

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts [hunk 1]
```diff
@@ -26,6 +29,27 @@ export type GatekeeperConfig = {
...
 };
 
+/**
+ * Create a new GatewayToken from an existing, setting the state to the given option state and optionally updating the expiry
+ * {GatewayToken} existingToken
+ * options.state: {State}
+ * options.expiry: {number}
+ * returns {GatewayToken}
+ */
+const tokenFromExisting = (
```
Comment (dankelleher) [inline solana/gatekeeper-lib/src/service/GatekeeperService.ts:39]: would this be better off as a function on GatewayToken?

```
const newGatewayToken = gatewayToken.update({ state, expiry })
```

**Review by dankelleher — COMMENTED** [review]
Loads of deleted files in this PR. is it correct? Seems unrelated to the change?

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts (hunk 1 above)
Comment (kevinhcolgan) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts:39]: gateway token properties are currently read-only on constructor so I'd have to change that but it's not a bad idea

Comment (kevinhcolgan) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts]: updated to add the 'update' function

**Comment by kevinhcolgan** [comment]
> Loads of deleted files in this PR. is it correct? Seems unrelated to the change?

gatekeeper http client was failing linting etc. and I didn't want to update an obsolete package - it's not used anymore

**Review by dankelleher — APPROVED** [review]

**Pull request merged**